distribution.pm

```diff
@@ -109,7 +110,12 @@ sub script_run {
     testapi::type_string "$cmd";
     if ($wait > 0) {
         my $str = testapi::hashed_string("SR$cmd$wait");
-        testapi::type_string " ; echo $str-\$?- > /dev/$testapi::serialdev\n";
+        if (testapi::is_serial_terminal) {
+            testapi::type_string " ; echo $str-\$?-\n";
+        }
+        else {
+            testapi::type_string " ; echo $str-\$?- > /dev/$testapi::serialdev\n";
+        }
```

Comment:
not so clean. If the implementations differ then this should be abstracted in the individual backends or consoles. So better provide a proposal of how to abstract that functionality in the testapi itself. The current tests use some redirection to the serial devices a lot. I assume if we can't abstract this away properly into os-autoinst this might be a hint that direct usage of a serial port is not a good idea to be combined with what os-autoinst is currently used for.